programming-languages/README.md

```diff
@@ -24,4 +24,29 @@
 ### Tools
 
 * [Compiler Explorer](https://godbolt.org/): Online C++ compiler. It lets you compile small code snippets and look at the generated assembly code. It supports different compilers across various architectures ranging from x86 to Arduinos.
-* [JetBrains CLion](https://www.jetbrains.com/clion/): One of the best C++ IDEs. CLion works on CMake making it easy to use with roscpp. 
\ No newline at end of file
+* [JetBrains CLion](https://www.jetbrains.com/clion/): One of the best C++ IDEs. CLion works on CMake making it easy to use with roscpp. 
+
+
+## Python
+A high level, interpreted, dead-simple, reads-like-english programming language used for a multitude of tasks, from Machine Learning/Data Analysis to deploying Full-Stack Websites.
+
+The [official website](https://www.python.org/) offers further resources, including talks, podcasts, conferences and the like.
+
+### Tutorials
+
+* [Sentdex Python Tutorial Series](https://www.youtube.com/playlist?list=PLQVvvaa0QuDe8XSftW-RAxdo6OmaeL85M): A collection of short videos that deal with everything from installing python, through loops, control flow, functions, upto usage of the package manager pip. Beyond this, the tutorial deals with specific use cases which may be skipped. It is highly recommended that you code along with the videos.
+
+
+### IDEs
+
+* [JetBrains' PyCharm](https://www.jetbrains.com/pycharm/): Fully featured Python IDE, offering powerful autocomplete, virtual environment support, comprehensive debugging, and notebook support. Users of other JetBrains IDEs will feel right at home.
+* [Jupyter Notebooks/ JupyterLab](https://jupyter.org/): Code in true interactive fashion, as you execute blocks and see outputs of unfinished code, which will help you code further. Runs in a browser, and a pip install is all that is required to get it.
+* [vscode](https://code.visualstudio.com/): A lightweight, extensible IDE with support for autocomplete, error-checking,git, ssh support and ipynb notebooks. It has a very powerful debugger and a long list of useful extensions/plugins. This is the favourite IDE of most of the developers and data scientists. 
+
+### Tools
+
+* **Environment/Dependency management**: [Conda](https://docs.conda.io/) or [venv](https://docs.python.org/3/library/venv.html) are typically used to isolate multiple python environments, so you don't end up with conflicting dependencies for different projects. This is highly recommended. It may seem like an inconvinience at first, but will save you a lot of time later on.
```

Comment:
(conda is recommended for ML and DL)

Reply:
Yes also add virtualenv as well

```
Virtualenv: A super lightweight tool for creating isolated virtual python environments. It doesn't require any setup and installs with a single command. It is also available for all platforms from raspberry to tegra based computer architecture. It is easy to use, portable and is arguably the best environment manager out there for any kind of python developers. 
```

Reply:
@take2rohit virtualenv is for Python2. venv is recommended for Python3 onwards. So, better not to recommend any Python2 tools.

Reply:
ok then we can add venv instead of virtualenv... both are almost same

Reply:
But strange enough i can use virtualenv for python3.8 env as well! @akshaykvnit how is this thing happening so?

Reply:
Sorry, I got confused between virtualenv and pyvenv. Pyvenv is deprecated. Whereas, virtualenv is a separate package from which venv was created as a standard Python package. So, venv should work for most simple use cases while virtualenv will help for more complicated stuff like having different python versions as well. You can add both virtualenv and venv. Mostly venv is useful because it's pre-installed and sometimes installing virtualenv globally may not be possible/allowed.

Reply:
@akshaykvnit we can do `pip3 install virtualenv` as well! Btw i never saw systems with venv pre-installed 🤔.

Reply:
Yeah, but pip install will install it systemwide which may not be desirable/allowed. Venv is actually an inbuilt library in python3. In any case, mention all possible options and people can figure out which suits them.

Reply:
venv comes with python by default, it was in the official wiki, which is why I added that. Virtualenv is an external, third party tool, so I didn't add that.

Reply:
Also, I recently switched to conda from virtualenv, turns out you don't have to do a systemwide CUDA install while using Conda, but that is required while using venv.